File: backend/app/schemas/media.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class MediaVariantRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    variant_name: str
    storage_key: str
    mime_type: str
    file_size: int
    width: int
    height: int
    created_at: datetime
    url: str | None = None


class MediaAssetRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    filename: str
    original_filename: str
    storage_key: str
    mime_type: str
    file_type: str
    file_size: int
    width: int | None
    height: int | None
    alt_text: str | None
    created_at: datetime
    updated_at: datetime
    variants: list[MediaVariantRead] = Field(
        default_factory=list,
    )
    url: str | None = None


class MediaAssetList(BaseModel):
    items: list[MediaAssetRead]
    total: int
    limit: int
    offset: int


class MediaAssetUpdate(BaseModel):
    alt_text: str | None = Field(
        default=None,
        max_length=500,
    )

File: admin-frontend/src/types/media.ts
export type MediaFileType = "image" | "document";

export interface MediaVariant {
  id: number;
  variant_name: string;
  storage_key: string;
  mime_type: string;
  file_size: number;
  width: number;
  height: number;
  created_at: string;
  url: string | null;
}

export interface MediaAsset {
  id: number;
  filename: string;
  original_filename: string;
  storage_key: string;
  mime_type: string;
  file_type: MediaFileType;
  file_size: number;
  width: number | null;
  height: number | null;
  alt_text: string | null;
  created_at: string;
  updated_at: string;
  variants: MediaVariant[];
  url: string | null;
}

export interface MediaAssetList {
  items: MediaAsset[];
  total: number;
  limit: number;
  offset: number;
}

export type MediaFilter = "all" | MediaFileType;
